app/models/loader/orchestrate.rb

```diff
@@ -118,6 +118,67 @@ def load
       store_passwords
 
       store_public_keys
+
+      emit_audit
+    end
+
+    SDID = Audit::SDID
+
+    OPS_MAP = {
+      "INSERT" => "add",
+      "DELETE" => "remove",
+      "UPDATE" => "change"
+    }
+
+    SUBJECT_MAP = {
+      :resource_id => 'resource'.freeze,
+      :role_id => 'role'.freeze,
+      :member_id => 'member'.freeze,
+      :name => 'annotation'.freeze,
+      :privilege => 'privilege'.freeze
+    }
+
+    def emit_audit
+      auth = policy_version.role.id
+      data = {
```

Comment:
Why is this just called `data`? I would like it to be easier to understand what's happening. Does this become the `subject`? If so, or if not, what's the connection between `data` and `subject`?

Reply:
`data` is the structured data to be output with the message. `subject` is the subject of this operation. `subject` field in the input gets translated to `subject@43868` SD-ID of the output.

Reply:
Please comment accordingly.